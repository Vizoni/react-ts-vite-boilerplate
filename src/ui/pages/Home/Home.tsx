export const Home = () => (
  <div>
    <h1>Nova home</h1>
  </div>
);
