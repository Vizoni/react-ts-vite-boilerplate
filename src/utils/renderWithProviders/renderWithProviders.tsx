import { render, type RenderOptions } from '@testing-library/react';
import { StrictMode, type ReactElement, type ReactNode } from 'react';

const Providers = ({ children }: { children: ReactNode }) => (
  <StrictMode>
    {/* <MyProvider {...props}>{children}</Myprovider>*/}
    {children}
  </StrictMode>
);

export const renderWithProviders = (
  ui: ReactElement,
  options?: RenderOptions
): ReturnType<typeof render> => render(ui, { wrapper: Providers, ...options });
