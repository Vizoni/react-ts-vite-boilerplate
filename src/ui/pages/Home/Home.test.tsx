import { renderWithProviders } from 'utils/renderWithProviders';

import { Home } from './Home';

const setup = () => renderWithProviders(<Home />);

describe('Home', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it('should match snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
