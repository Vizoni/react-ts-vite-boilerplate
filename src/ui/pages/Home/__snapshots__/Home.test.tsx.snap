// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<div>
  <div>
    <h1>
      Nova home
    </h1>
  </div>
</div>
`;
